socorro/signature/rules.py

```diff
@@ -418,7 +423,7 @@ def __init__(self):
     def _create_frame_list(self, crashing_thread_mapping, make_modules_lower_case=False):
         frame_signatures_list = []
         for a_frame in islice(
-            crashing_thread_mapping.get('frames', {}),
+            crashing_thread_mapping.get('frames', []),
```

Comment:
Frames is a list of things--it should default to an empty list and not an empty dict.